### ISSUE TITLE
Try ECCOv4-r4 on ARCHER2 full system

### ISSUE THREAD

**Issue by DaniJonesOcean** [eCSE-MITgcm-ARCHER2/eCSE-archer2-eccov4r4#1]
- Try following the installation instructions on the webpage here: https://docs.archer2.ac.uk/research-software/mitgcm/mitgcm/
- Update the webpage as needed (via pull requests). 
- Try the ECCOv4 setup with 360 cores as well
- Meet with Mike M. about the results, next steps for contributing optimized build options file to the MITgcm GitHub repo

As part of this, address this specific issue:
https://github.com/ARCHER2-HPC/archer2-docs/issues/294

**Comment by DaniJonesOcean** [comment]
From Mike M:

>>> /work/n02/shared/mjmn02/ECCOv4/cases/cce/cce1/scripts/ dev_linux_amd64_cray_archer2
And
/work/n02/shared/mjmn02/ECCOv4/cases/valid/gnu/gnu_O3unr/scripts/ dev_linux_amd64_gfortran_archer2
For 360 cores it seem natural to start with 3 nodes, 120 each.

**Comment by DaniJonesOcean** [comment]
Specific things:

> cp /work/n02/shared/mjmn02/ECCOv4/cases/cce/cce1/scripts/dev_linux_amd64_cray_archer2 MITgcm/tools/build_options/

And here are some lines for the environment variables:

>>> export MITGCM_ROOTDIR=/mnt/lustre/a2fs-work1/work/n01/n01/dannes1/eCSE_test/MITgcm
export PATH=$MITGCM_ROOTDIR/tools:$PATH
export MITGCM_OPT=$MITGCM_ROOTDIR/tools/build_options/dev_linux_amd64_cray_archer2

I'm having an issue with NetCDF modules at present. They're not loading. Probably not needed for compiling and running, but would definitely be needed eventually.

**Comment by DaniJonesOcean** [comment]
Make a note to change:

> genmake2 -mods /path/to/additional/source -mpi -optfile $MITGCM_OPT

to 

> genmake2 -mods ../code -mpi -optfile $MITGCM_OPT

**Comment by DaniJonesOcean** [comment]
Okay, ECCOv4-r4 compiles fine on the full system. Will need to wait on the forcing dataset to run it

**Comment by DaniJonesOcean** [comment]
It compiles and runs! Yay!

I've tried it with the cray compiler. Now to try it with the gnu compiler. And possibly change the documentation to better reflect using the two files.

**Comment by DaniJonesOcean** [comment]
Also need to get NetCDF working. Some things are breaking, which may be due to this.

**Comment by DaniJonesOcean** [comment]
Weirdly, at runtime, MITgcm complains that it can'd find the initial hydrography files. It's not even using those files, I don't think. I believe it's using the xx files instead. 

r3 also didn't have these files. What code is catching it? Did we not download the correct MITgcm checkpoint?

**Comment by DaniJonesOcean** [comment]
Bizarrely, I had to comment out the initial hydrography lines in `data` in order to get the code to run. Strange.

**Comment by DaniJonesOcean** [comment]
Closing this issue. (I think I just forgot to earlier. These instructions seem to work fine.)
